lib/types/index.ts

```diff
@@ -1,10 +1,12 @@
 
 import { Readable } from 'stream';
-import { AbstractLevelDOWN } from 'abstract-leveldown'
+import { AbstractChainedBatch, AbstractLevelDOWN } from 'abstract-leveldown'
 import {AsyncIterator} from 'asynciterator';
 import {Literal, DataFactory, Quad_Subject, Quad_Predicate, Quad_Object, Quad_Graph, Quad, Term} from 'rdf-js';
 
-export interface EmptyOpts {}
+export interface BatchOpts {
+  preWrite?: (batch: AbstractChainedBatch) => Promise<unknown> | void;
```

Comment:
We can `await` non-promise values, so this can become `=> any` I think.

Reply:
I made this explicit so it's self-documenting that you can be async. (I'll add a comment anyway.)

Reply:
Makes sense! I like the approach. Just change it to `Promise<any> | any` for maximum compatibility.